praw/models/user.py

```diff
@@ -117,7 +117,7 @@ def me(
     ) -> Optional[Redditor]:  # pylint: disable=invalid-name
         """Return a :class:`.Redditor` instance for the authenticated user.
 
-        In :attr:`~praw.Reddit.read_only` mode, this method returns ``None``.
+        It is an error to call this method in :attr:`~praw.Reddit.read_only` mode.
```

Comment:
Don't you think it would be better to use the `:raises: ...` syntax or not?

Reply:
I think `:raises:` expects you to document a particular exception. My concern is that if an exception type is documented then users will take that as an indication to catch and handle it. It’s just like how we shouldn’t document `MemoryError` for every function that creates an object.

Reply:
What's stopping them from doing so anyways? There's no way of knowing, in like a end-user program, if the user is running in auth mode or not in auth mode. Then it would be a good idea to handle those exceptions.

Reply:
Nothing’s stopping them. If they want to catch an exception instead of write a cheap check before doing the action then they are more than welcome to do so. It’s simply not preferred.

Not sure what you mean by ‘there’s no way of knowing’ if the reddit client is in ‘auth mode’. The way is `reddit.read_only`.